src/main/java/ru/l1ttleO/boyfriend/commands/util/CommandReader.java

```diff
@@ -0,0 +1,118 @@
+package ru.l1ttleO.boyfriend.commands.util;
+
+import java.util.Iterator;
+import java.util.List;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.User;
+import org.jetbrains.annotations.NotNull;
+import ru.l1ttleO.boyfriend.I18n.BotLocale;
+import ru.l1ttleO.boyfriend.Utils;
+import ru.l1ttleO.boyfriend.exceptions.WrongUsageException;
+import ru.l1ttleO.boyfriend.settings.Settings;
+
+import static ru.l1ttleO.boyfriend.I18n.tl;
+
+public class CommandReader {
+    public final @NotNull String prefix;
+    public final @NotNull String alias;
+    @Deprecated
+    public final @NotNull String[] args;
+    private final @NotNull Iterator<String> iter;
+    public final @NotNull Sender sender;
+
+    public CommandReader(final @NotNull String prefix, final @NotNull String input, final @NotNull Sender sender) {
+        this.prefix = prefix;
+        this.args = input.split(" ");
+        this.iter = List.of(this.args).iterator();
+        this.alias = this.iter.next().substring(prefix.length());
+        this.sender = sender;
+    }
+
+    public WrongUsageException noArgumentException(final @NotNull String argumentName) throws WrongUsageException {
```

Comment:
```suggestion
    public WrongUsageException noArgumentException(final @NotNull String argumentName) {
```
`The declared exception 'WrongUsageException' is never thrown` [intellij]